### PR TITLE
Use ServiceWorkerGlobalScope as fallback for Window

### PR DESCRIPTION
**Pull request by kjvalencik**
Cherry-pick of https://github.com/oliverdunk/ring/commit/d92ad9cb363cca8d3372d75a02ae3fc77a2694a2

CC: @oliverdunk

### CODE REVIEW

**Review by oliverdunk — DISMISSED** [review]
Thanks for this!

File: Cargo.toml [hunk 1]
```diff
@@ -340,7 +340,9 @@ once_cell = { version = "1.5.2", default-features = false, features=["std"], opt
 once_cell = { version = "1.5.2", default-features = false, features=["std"] }
 
 [target.'cfg(all(target_arch = "wasm32", target_vendor = "unknown", target_os = "unknown", target_env = ""))'.dependencies]
-web-sys = { version = "0.3.37", default-features = false, features = ["Crypto", "Window"] }
+web-sys = { version = "0.3.51", default-features = false, features = ["Crypto", "Window", "ServiceWorkerGlobalScope"], optional = true }
```
Comment (complexspaces) [inline Cargo.toml]: Why was this made optional?

Comment (oliverdunk) [reply Cargo.toml]: Looks like in my commit it was optional both before and after 🤔. I guess something changed there and the merge went bad.

Comment (kjvalencik) [reply Cargo.toml]: It's optional on `main`. I think I made a mistake in the merge conflict.

Comment (kjvalencik) [reply Cargo.toml]: Fixed.

**Review by complexspaces — APPROVED** [review]

**Pull request merged**